projects/addon-editor/components/editor-socket/editor-socket.component.ts

```diff
@@ -44,5 +47,32 @@ export class TuiEditorSocketComponent {
         @Optional()
         @Inject(TUI_SANITIZER)
         private readonly tuiSanitizer: Sanitizer | null,
+        @Inject(DOCUMENT)
+        private readonly document: Document,
+        @Optional()
+        @Inject(TuiTiptapEditorDirective)
+        private readonly editor: TuiTiptapEditorDirective | null,
     ) {}
+
+    /**
+     * @description:
+     * the main problem is that the external environment editor can use different base href="../"
+     * More information: https://rogerkeays.com/blog/using-base-href-with-anchors
+     */
+    @HostListener(`click`, [`$event`])
+    click(event: Event): void {
+        if (this.editor) {
```

Comment:
Can we do something like `if (this.editor || !tuiIsElement(event.target)) {` so that we don't need to typecast on line 68?

Reply:
fixed